examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""
+====================================
+Demo of HDBSCAN clustering algorithm
+====================================
+
+"""
+
+import numpy as np
+
+from sklearn.cluster import HDBSCAN
+from sklearn import metrics
+from sklearn.datasets import make_blobs
+from sklearn.preprocessing import StandardScaler
+
+
+# %%
+# Generate sample data
+# --------------------
+centers = [[1, 1], [-1, -1], [1, -1]]
+X, labels_true = make_blobs(
+    n_samples=750, centers=centers, cluster_std=0.4, random_state=0
+)
+
+X = StandardScaler().fit_transform(X)
+
+# %%
+# Compute HDBSCAN
+# ---------------
+KWARGS = ({}, {"min_samples": 2}, {"min_cluster_size": 25})
+models = []
+
+for kwargs in KWARGS:
+    hdb = HDBSCAN(**kwargs).fit(X)
+    models.append((hdb.labels_, hdb.probabilities_, kwargs))
+    labels = hdb.labels_
+
+    # Number of clusters in labels, ignoring noise
+    n_clusters_ = len(set(labels)) - (1 if -1 in labels else 0)
+    n_noise_ = list(labels).count(-1)
+
+    print(f"\nFor {kwargs=}")
+    print("Estimated number of clusters: %d" % n_clusters_)
+    print("Estimated number of noise points: %d" % n_noise_)
+    print("Homogeneity: %0.3f" % metrics.homogeneity_score(labels_true, labels))
+    print("Completeness: %0.3f" % metrics.completeness_score(labels_true, labels))
+    print("V-measure: %0.3f" % metrics.v_measure_score(labels_true, labels))
+    print(
+        "Adjusted Rand Index: %0.3f" % metrics.adjusted_rand_score(labels_true, labels)
+    )
+    print(
+        "Adjusted Mutual Information: %0.3f"
+        % metrics.adjusted_mutual_info_score(labels_true, labels)
+    )
+    print("Silhouette Coefficient: %0.3f" % metrics.silhouette_score(X, labels))
```

Comment:
I think we can make this example more narrative driven, rather that listing out metrics.  To me one of the more interesting ideas around HDBSCAN is how it is like doing a hyperparameter search over eps in DBSCAN. In an example, we can do the hyperparameter search in DBSCAN and compare it to one run of HDBSCAN.

The outline would like:

1. Use HDBSCAN and plot the cluster and non-cluster points
2. Update `min_samples` and explain what happens to the cluster.
3. Do a hyperparameter search over eps with DBSCAN and time it.
4. Plot the clustering with the most optimal eps for DBSCAN and HDBSCAN

Reply:
I overhauled the plotting example, let me know what you think.